### ISSUE TITLE
Not linting files in a symlinked folder using a local install of eslint

### ISSUE THREAD

**Issue by jbmusso** [roadhump/SublimeLinter-eslint#57]
(Sorry, I pressed cmd+enter too fast by accident. Updated post.)

Files present in a symlinked subfolder are not being linted by SublimeLinter-eslint.

Use case: I symlinked a couple subfolders of my application into `node_modules` in order to be able to do `require('subfolder')` instead of doing `require('../../../../subfolder')` (using browserify, see https://github.com/substack/browserify-handbook#avoiding-).

When using a local install of `eslint`, SublimeLinter-eslint doesn't lint files in these symlinked subfolders. A workaround is to symlink `/myapp/node_modules/.bin` as `/myapp/my/subfolder/node_modules/.bin` in order to have the local `eslint` bin available for that `subfolder` ahem.. folder. This is a workaround, I'm unsure if there's a better way to handle this specific case when using a local install of eslint.

**Comment by roadhump** [comment]
I change plugin to support ignore setting of ESLint. By default files in `node_modules` folder are ignored. You can try to toggle SublimeLinter debug mode and open SublimeText console to check if output of ESLint confirms it.

**Comment by mclapa** [comment]
Hello

I have a similar problem.

I noticed it only with eslint (which I prefer) but have switched to jsxhint as it doesn't throw the same warning

Essentially I'm mapping a remote drive to my local drive (sshfs) and for some reason (just with eslint) it is not linting any of the files, the following error 

SublimeLinter: eslint: myfile.jsx ['/usr/local/bin/eslint', '--format', 'compact', '--stdin', '--stdin-filename', '__RELATIVE_TO_FOLDER__'] 
SublimeLinter: eslint output:
Invalid option '--stdin-filename' - perhaps you meant '--stdin'?

**Comment by roadhump** [comment]
@mclapa, update eslint, SublimeLinter should check version but sometimes it does not

**Comment by martinhbramwell** [comment]
I believe this issue should **not** be closed.  I have linting on files recursively within an opened folder, but not on files of external folders referred to by symbolic links within the opened folder.

I am running a fresh install of Xubuntu 14.04

I install eslint *(v1.3.1)* with npm *(v1.4.28)* : 
```
npm install -gy eslint
npm install -gy eslint-plugin-react
npm install -gy babel-eslint
```

I install ST3 *(Stable Channel, Build 3083)* with : 
```
add-apt-repository -y ppa:webupd8team/sublime-text-3
apt-get update
apt-get install -y sublime-text-installer
```

I install linting in ST3 with Package Control >> Install Package :  
```
SublimeLinter
SublimeLinter-contrib-eslint
```

Is there something more I need to do?

The opened folder contains : 
```
yourself@yourpc:~/projects/YourApp$ tree -L 2
.
├── example_circle.yml
├── LICENSE
├── master.zip
├── packages
│   └── yourpackage -> /home/yourself/projects/packages/yourself/yourpackage
├── README.md
├── tests
│   └── tinyTests
├── YourApp.css
├── YourApp.html
└── YourApp.js

4 directories, 7 files
```

The symlinked folder looks like this : 
```
yourself@yourpc:~/projects/packages/yourself/yourpackage$ tree -L 2
.
├── package.js
├── README.md
├── yourpackage.js
└── yourpackage-tests.js

0 directories, 4 files
yourself@yourpc:~/projects/packages/yourself/yourpackage$ 
```

I get linting on ```YourApp.js```, but not on ```yourpackage-tests.js```.

**Comment by martinhbramwell** [comment]
Ok, I have placed a ```.eslintrc``` file in the symlinked directory and it works.

It would be nice to have a switch in ```SublimeLinter.sublime-settings``` to choose whether or not to use parent folder's ```.eslintrc```.

**Comment by roadhump** [comment]
I investigated this issue, and it seems real. When plugin runs, ESLint resolves file path as real path, not path in project by linked folder. It can be solved by set `--stdin-filename` to `@` as here https://github.com/roadhump/SublimeLinter-eslint#contextgetfilename-in-rule-returns-relative-path
. 

I'll add this issue to README.
